src/bp-messages/classes/class-bp-messages-message.php

```diff
@@ -277,13 +277,18 @@ public static function get_recipient_ids( $recipient_usernames ) {
 	 * @return int|null ID of the message if found, otherwise null.
 	 */
 	public static function get_last_sent_for_user( $thread_id ) {
-		global $wpdb;
-
-		$bp = buddypress();
 
-		$query = $wpdb->get_var( $wpdb->prepare( "SELECT id FROM {$bp->messages->table_name_messages} WHERE sender_id = %d AND thread_id = %d ORDER BY date_sent DESC LIMIT 1", bp_loggedin_user_id(), $thread_id ) );
+		$query = self::get(
+			array(
+				'fields'          => 'ids',
+				'user_id'         => bp_loggedin_user_id(),
+				'include_threads' => array( $thread_id ),
+				'page'            => 1,
+				'per_page'        => 1,
+			)
+		);
 
-		return is_numeric( $query ) ? (int) $query : $query;
+		return is_numeric( $query['messages'][0] ) ? (int) $query['messages'][0] : $query['messages'][0];
```

Comment:
@janakkaneriya007 Can you check $wpdb->get_var() function which value return is result is empty. we need to return same value in else part.